nni/retiarii/execution/api.py

```diff
@@ -67,3 +69,8 @@ def query_available_resources() -> int:
 
 def is_stopped_exec(model: Model) -> bool:
     return model.status in (ModelStatus.Trained, ModelStatus.Failed)
+
+
+def budget_exhausted() -> bool:
+    advisor = get_advisor()
```

Comment:
Suggest access to execution engine rather than directly to advisor.

Suggest rename budget to `resource` for consistency

Reply:
`resource` is a little misleading, because in our current implementation, resource means concurrency